### PR TITLE
V0.15.0 RC -- DO NOT MERGE!

### PR DESCRIPTION
**Pull request by rhornung67**
# Summary

- This PR is to compare changes for the upcoming RAJA release to the develop branch. Please review this AND APPROVE the PR to merge into main: https://github.com/LLNL/RAJA/pull/1222

### CODE REVIEW

File: scripts/spack_packages/raja/package.py [hunk 1]
```diff
@@ -182,9 +185,9 @@ def hostconfig(self, spec, prefix, py_site_pkgs_dir=None):
         cfg.write("###################\n".format("#" * 60))
         cfg.write("# Generated host-config - Edit at own risk!\n")
         cfg.write("###################\n".format("#" * 60))
-        cfg.write("# Copyright (c) 2020, Lawrence Livermore National Security, LLC and\n")
-        cfg.write("# other Umpire Project Developers. See the top-level LICENSE file for\n")
-        cfg.write("# details.\n")
+        cfg.write("# Copyright 2016-21, Lawrence Livermore National Security, LLC\n")
```
Comment (rchen20) [inline scripts/spack_packages/raja/package.py:188]:
```suggestion
        cfg.write("# Copyright 2016-22, Lawrence Livermore National Security, LLC\n")
```

File: RELEASE_NOTES.md [hunk 2]
```diff
@@ -9,6 +9,131 @@
...
+      * BREAKING CHANGE: RAJA OffsetLayout constructor was changed to take
+        (begin, end) args (where end is one past the last index) instead of
+        (first, last) args (where last index was included). This is consistent 
+        with expected behavior and other RAJA Layout/View concepts.
+      * New experimental features that support SIMD/SIMT programming by 
+        guaranteeing vectorization without the need to rely on compiler
+        auto-vectorization. Basic documentation for this is included in the
+        RAJA User Guide and should provide enough description for interested
+        users to try it out. 
+      * "Flatten" policies were added for RAJA Teams. This reshaping
```
Comment (MrBurmark) [inline RELEASE_NOTES.md:53]:
```suggestion
      * "Flatten" policies were added for RAJA Teams. This reshapes
```

File: RELEASE_NOTES.md [hunk 3]
```diff
@@ -9,6 +9,131 @@
...
+      * New experimental features that support SIMD/SIMT programming by 
+        guaranteeing vectorization without the need to rely on compiler
+        auto-vectorization. Basic documentation for this is included in the
+        RAJA User Guide and should provide enough description for interested
+        users to try it out. 
+      * "Flatten" policies were added for RAJA Teams. This reshaping 
+        multi-dimensional GPU thread blocks to 1D.
+      * RAJA Teams now allows a single execution policy to be provided. 
+        Previously, it required two; e.g., a CPU policy and a GPU policy.
+      * ROCTX suport has been added to enable kernel naming with RAJA Teams.
```
Comment (MrBurmark) [inline RELEASE_NOTES.md:57]:
```suggestion
      * ROCTX support has been added to enable kernel naming with RAJA Teams.
```

File: RELEASE_NOTES.md [hunk 4]
```diff
@@ -9,6 +9,131 @@
...
+      * Details of CUDA and HIP errors are now added to the reported exception
+        string. Previously, this information was going to stderr.
+      * All CUDA execution policies have been expanded to allow users to specify
+        a minimum number of blocks per SM, if they wish to do that. An analogous
+        capability for HIP execution policies is being hashed out. 
+      * Changes were made to RAJA scans to address a consistency issue and
+        allow an array corresponding to an input span to passed as
+        a const pointer.
```
Comment (MrBurmark) [inline RELEASE_NOTES.md:65]:
```suggestion
        allow const pointers to be passed as the input span.
```

File: RELEASE_NOTES.md [hunk 5]
```diff
@@ -9,6 +9,131 @@
...
+         * The naming scheme for SYCL 'group' and 'local' policies has been
+           changed from 1-based to 0-based for block dimensions.
+         * The use of the SYCL atomic OneAPI extension namespace has been 
+           cleaned up.
+
+  * Build changes/improvements:
+      * AS OF THIS RELEASE, RAJA REQUIRES A C++14-COMPLIANT COMPILER TO BUILD!! 
+      * AS OF THIS RELEASE, RAJA REQUIRES CMAKE version 3.14.5 or newer.
+      * The BLT submodule is updated to v0.5.0, which includes improved
+        support for ROCm/HIP builds.Although the option CMAKE_HIP_ARCHITECTURES
```
Comment (MrBurmark) [inline RELEASE_NOTES.md:93]:
```suggestion
        support for ROCm/HIP builds. Although the option CMAKE_HIP_ARCHITECTURES
```

File: RELEASE_NOTES.md [hunk 6]
```diff
@@ -9,6 +9,131 @@
...
+      * CMake logic for using CUB in RAJA for a CUDA build has been changed.
+        The default behavior is now that when the CUDA version is < 11, the
+        RAJA CUB submodule will be used. When the CUDA version is >= 11, the
+        CUB version that is included in the associated CUDA toolkit will be 
+        used. Users have the ability to override these defaults and select
+        a specific version of CUB if they wish.
+      * CMake logic for using rocPRIM in RAJA for a HIP build is similar.
+        The default behavior is now that when the HIP version is < 4, the
+        RAJA rocPRIM submodule will be used. When the HIP version is >= 4, the
+        rocPRIM version that is included in the associated ROCmM toolkit will
```
Comment (MrBurmark) [inline RELEASE_NOTES.md:120]:
```suggestion
        rocPRIM version that is included in the associated ROCM toolkit will 
```

Comment (rhornung67) [reply RELEASE_NOTES.md:120]: Fixed.

File: RELEASE_NOTES.md [hunk 7]
```diff
@@ -9,6 +9,131 @@
...
+      * The camp submodule is updated to v0.5.3. If you do not use the submodule
+        and build RAJA with an external version of camp, you will need to
+        use camp v0.5.3 or newer.
+      * The "RAJA_" prefix has been added to all CMake options. Options that 
+        shadow a CMake or BLT option are turned into cmake_dependent_option 
+        calls, ensuring that they can be controlled independently and have the 
+        correct dependence on the underlying CMake or BLT support;
+        e.g., RAJA_ENABLE_CUDA requires ENABLE_CUDA.
+      * The camp_DIR export has been removed. Camp paths will be searched 
+        using the default logic which consistent with camp.
```
[the remark this answers is not in the document]
Comment (rhornung67) [reply RELEASE_NOTES.md:106]: fixed

File: RELEASE_NOTES.md (hunk 4 above)
Comment (rhornung67) [reply RELEASE_NOTES.md:65]: fixed.

File: RELEASE_NOTES.md (hunk 3 above)
Comment (rhornung67) [reply RELEASE_NOTES.md:57]: fixed

File: RELEASE_NOTES.md (hunk 2 above)
Comment (rhornung67) [reply RELEASE_NOTES.md:53]: fixed

**Review by trws — APPROVED** [review]

**Review by rchen20 — APPROVED** [review]

**Pull request closed without merging**